File: Program's_Contributed_By_Contributors/Python_Programs/Insertion_Sort.py
# Recursive Insertion sort 
def insertionSort(arr,n):
   # base case
   if n<=1:
      return
   # Recursive Call
   insertionSort(arr,n-1)
   key = arr[n-1]
   j = n-2
   # Compare key with each element on the left of it until an element smaller than it is found
   while (j>=0 and arr[j]>key):
      arr[j+1] = arr[j]
      j = j-1
   # Place key at after the element just smaller than it.
   arr[j+1]=key
a=[]
n=int(input("Enter number of elements: "))
for i in range(0,n):
    b=int(input("Enter element: "))
    a.append(b)
insertionSort(a,n)
print('Sorted Array:')
print(a)

File: Program's_Contributed_By_Contributors/C++ Programs/Array/InsertionSort.cpp
#include <iostream>
using namespace std;

void printing(int arr[],int n){
    for (int i = 0; i < n; i++)
    {
        cout << arr[i] << " ";
    }   
}

int main()
{
    int n;
    cout<<"Enter the number of elements in array-";
    cin >> n;
    int arr[n];
    for (int i = 0; i < n; i++)
    {
        cin >> arr[i];
    }
    cout<<"\n\nUnsorted array is"<<endl;
    printing(arr,n);
    for (int i = 0; i < n; i++)
    {
        int curr = arr[i];
        int j = i-1;
        while(arr[j] > curr && j!=-1){
            arr[j+1]=arr[j];
            j--;
        }
        arr[j+1]=curr;
    }
    cout<<"\n\nSorted array is"<<endl;
    printing(arr,n);
    return 0;
}